src/library/scanner/libraryscanner.cpp

```diff
@@ -170,6 +156,17 @@ void LibraryScanner::run() {
 
 void LibraryScanner::slotStartScan() {
     qDebug() << "LibraryScanner::slotStartScan";
+    DEBUG_ASSERT(m_state == STARTING);
```

Comment:
Yes, compile-time error is the wrong word. 
You can only change the value of m_state here, by recompiling Mixxx.